### DISCUSSION TITLE
Consider providing a NuGet credential provider plugin for GitHub Packages

### DISCUSSION THREAD

**Discussion by PathogenDavid** [community/community#128462] [Packages]
### Select Topic Area

Product Feedback

### Body

It would be great it GitHub Packages provided a [NuGet credential provider](https://learn.microsoft.com/en-us/nuget/consume-packages/consuming-packages-authenticated-feeds#credential-providers) similar to what's provided by Azure Artifacts and such.

Right now the client experience with GitHub Packages is fairly subpar unless someone has manually configured their credentials. Visual Studio throws up a [mysterious password dialog](https://github.com/bonsai-rx/bonsai/assets/278957/3cf2e041-a58a-4018-a946-3e924e152fd9) (which is misleading since you need to provide a PAT rather than a password) and interacting with the feed via the `dotnet` commandline simply fails.

This would make it much more practical to reference a GitHub Packages feed within a `NuGet.config` saved in the repo (where hard-coding someone's PAT would be unacceptable.)

**Comment by grandmas-favorite** [comment]
Hi @PathogenDavid!

We appreciate your feedback and your valuable input. We want to be transparent about our current priorities. You can gain insights into the projects we’re currently focusing on for the upcoming year by checking our [public roadmap](https://github.com/github/roadmap/projects/1). While smaller fixes may not always be explicitly listed, they often get addressed when related work is in progress.

Regarding your specific request, it’s important to note that it’s not actively being worked on at the moment. However, this doesn’t mean we won’t consider it in the future; it’s simply not on our development roadmap right now.

We’d like to emphasize that as a valued GitHub user, your feedback in this Discussion space, along with appropriate tags and examples, plays a crucial role in shaping the development of GitHub. While we can’t guarantee immediate action on every piece of feedback, please rest assured that we are attentive to your suggestions and continuously strive to enhance the GitHub experience.

While we do review all incoming feedback unfortunately due to the volume of feedback, we may not be able to respond to each one individually at this time. However, we strongly encourage you to continue providing your valuable input as it greatly helps us prioritize and improve GitHub.

Your contributions are instrumental in making GitHub better for all users. Thank you for taking the time to share your thoughts with us, even if the particular request is not currently in active development. Your input is truly valuable and contributes to the continuous improvement of GitHub.

**Reply by Santas** [reply]
It is bizarre that Github as part of Microsoft has subpar support for .NET related tooling.

**Reply by smokedlinq** [reply]
If you expect us to be able to migrate out of Azure DevOps fully (I would like to be able to anyway), packages need to support the developer experience securely, and PATs are not the way.
